jvm/src/main/scala/async/JvmAsyncOperations.scala

```diff
@@ -1,12 +1,17 @@
 package gears.async
 
 object JvmAsyncOperations extends AsyncOperations:
+  override def sleep(millis: Long)(using Async): Unit =
+    jvmInterruptible(Thread.sleep(millis))
 
-  private def jvmInterruptible[T](fn: => T)(using Async): T =
+  /** Runs `fn` in a [[cancellationScope]] where it will be interrupted (as a Java thread) upon cancellation.
+    *
+    * Note that `fn` will need to handle both [[java.util.concurrent.CancellationException]] (when performing Gears
+    * operations such as `.await`) *and* [[java.lang.InterruptedException]], so the intended use case is usually to wrap
+    * interruptible Java operations, containing `fn` to a narrow scope.
+    */
+  def jvmInterruptible[T](fn: => T)(using Async): T =
```

Comment:
We already don't compile this function on an unsupported platform. I usually don't like the idea that we expose a function that sounds like it does something but is actually a no-op underneath :sweat: 

We could call this function `interruptible` since it's already under `JvmAsyncOperations`, but I think it might be confused with the `uninterruptible` function that we *do* have outside. 
They also take the same argument types, to add to the fire :)